subprojects/dependency-management/src/main/java/org/gradle/api/internal/artifacts/dsl/dependencies/DefaultDependencyHandler.java

```diff
@@ -345,7 +347,9 @@ public Dependency enforcedPlatform(Object notation) {
         Dependency platformDependency = create(notation);
         if (platformDependency instanceof ExternalModuleDependency) {
             ExternalModuleDependency externalModuleDependency = (ExternalModuleDependency) platformDependency;
-            DeprecationLogger.whileDisabled(() -> externalModuleDependency.setForce(true));
```

Comment:
I do not think this will end up preserving semantics. Can be verified by enabling back the tests mentioned above.

We might need a new internal API to represent this.